File: src.JAVA/src/swa/ariadne/ctrl/IAriadneEventHandler.java
package swa.ariadne.ctrl;

/**
 * The {@link AriadneController} will call these methods on a MazeForm or other client.
 * 
 * @author Stephan.Wacker@web.de
 */
public interface IAriadneEventHandler
{
    /**
     * Called before processing every timer event.
     * @return False if the event is to be ignored.
     */
    boolean Alive();

    /**
     * Creates a new (different) maze.
     */
    void OnNew();

    /// <summary>
    /// Called when the current maze has been solved.
    /// </summary>
    // TODO: void PrepareForNextStart();

    /// <summary>
    /// Called when the controller has been reset.
    /// </summary>
    // TODO: void ResetCounters();
    
    /// <summary>
    /// Called for updating the display before a time consuming operation.
    /// </summary>
    // TODO: void Update();
}

File: src.CS/SWA.Ariadne.Ctrl/IAriadneEventHandler.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SWA.Ariadne.Ctrl
{
    /// <summary>
    /// The AriadneController will call these methods on a MazeForm or other client.
    /// </summary>
    public interface IAriadneEventHandler
    {
        /// <summary>
        /// Called before processing every timer event.
        /// Should return false if the event is to be ignored.
        /// </summary>
        bool Alive { get; }

        /// <summary>
        /// Called when the controller's state has changed.
        /// </summary>
        void NotifyControllerStateChanged();

        /// <summary>
        /// Called when starting a new run.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        void OnNew(object sender, EventArgs e);

        /// <summary>
        /// Called when the current maze has been solved.
        /// </summary>
        void PrepareForNextStart();

        /// <summary>
        /// Called when the controller has been reset.
        /// </summary>
        void ResetCounters();
        
        /// <summary>
        /// Called for updating the display before a time consuming operation.
        /// </summary>
        void Update();
    }
}
